parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java

```diff
@@ -371,4 +371,32 @@ public boolean equals(Object object) {
     }
     return false;
   }
+
+  private boolean checkCompatibility(BlockSplitBloomFilter that) {
+    return this.maximumBytes == that.maximumBytes && this.hashStrategy == that.hashStrategy;
+  }
+
+  @Override
+  public BloomFilter union(BloomFilter bloomFilter) {
+    if (bloomFilter instanceof BlockSplitBloomFilter) {
+      BlockSplitBloomFilter that = (BlockSplitBloomFilter) bloomFilter;
+
+      if (!checkCompatibility(that)) {
+        throw new RuntimeException("Bloom filters are not compatible!");
+      }
+
+      int numBytes = this.bitset.length;
+      byte[] newBitset = new byte[numBytes];
```

Comment:
Do we need to allocate a new buffer, why not use `this.bitset[index] |= that.bitset[index]` instead?

Reply:
I agree!